README.md

```diff
@@ -19,141 +20,189 @@ When the reader has completed this Code Pattern, they will understand how to:
+* Having said that, this is only for this tutorial purpose, and you can define your Custom ML provider endpoint in any fashion you want, such that it wraps your own custom ML engine.
+* The scoring request and response payload should confirm to the schema as described here at: https://dataplatform.cloud.ibm.com/docs/content/wsj/model/wos-frameworks-custom.html
+* To start the application using the below code, make sure you install the python packages using:
 
+```python
 
-* Wait a couple of minutes for the database to be provisioned.
-* Click on the `Service Credentials` tab on the left and then click `New credential +` to create the service credentials. Copy them or leave the tab open to use later in the notebook.
+pip install -r requirement.txt
+```
 
-### 3. Create a notebook in IBM Watson Studio
+</details>
 
-* In [Watson Studio](https://dataplatform.cloud.ibm.com/), create a `New project`.
-* Using the project you've created, click on `+ Add to project` and then choose the  `Notebook` tile, OR in the `Assets` tab under `Notebooks` choose `+ New notebook` to create a notebook.
-* Select the `From URL` tab.
-* Enter a name for the notebook.
-* Optionally, enter a description for the notebook.
-* Under `Notebook URL` provide the following url: https://raw.githubusercontent.com/IBM/monitor-custom-ml-engine-with-watson-openscale/master/notebooks/WatsonOpenScaleAndCustomMLEngine.ipynb
-* Select the `Default Python 3.5` runtime, either `Free` or `XS`.
-* Click the `Create` button.
+or 
 
-### 4a. Deploy to IBM Cloud
+<details><summary>Run on RedHat OpenShift</summary>
 
-[![Deploy to IBM Cloud](https://cloud.ibm.com/devops/setup/deploy/button.png)](https://cloud.ibm.com/devops/setup/deploy?repository=https://github.com/IBM/monitor-custom-ml-engine-with-watson-openscale)
+## Prerequisites
 
-Click the ``Deploy to IBM Cloud`` button and hit ``Create`` and then jump to step 5.
+You will need a running OpenShift cluster, or OKD cluster. You can provision [OpenShift on the IBM Cloud](https://cloud.ibm.com/kubernetes/catalog/openshiftcluster).
 
-**OR**
+## Steps
 
-### 4.b Run the application server locally
+1. [Create an OpenShift project](#1-create-an-openshift-project)
+1. [Create the config map](#2-create-the-config-map)
+1. [Get a secure endpoint](#3-get-a-secure-endpoint)
+1. [Run the web app](#4-run-the-web-app)
 
-> NOTE: Running locally will require Python 3.5 or 3.6 (later versions will not work with Tensorflow).
-If you run the server locally, it may not have a publicly addressible IP address, if you are behind a firewall or local router. You would therefore also have to run the [Jupyter notebook](https://jupyter.org/) locally as well.
+### 1. Create an OpenShift project
 
-* It is recommended that you use a [Python virtualenv](https://pypi.org/project/virtualenv/)
+* Using the OpenShift web console, select the `Application Console` view.
```

Comment:
what version of OpenShift is this? Sure these screens still exist?

Reply:
Not sure. This image is from Nov 1 (you added). I haven't updated.